### ISSUE TITLE
Fake study

### ISSUE THREAD

**Issue by AHMEDFOUAAD** [AzazHassankhan/Machine_Learning_based_Trading_Techiques#1]
The code is using future data in the model training features,".shift(-1)" which is leading to look ahead bias:
df['Open-Close']=(df["open"]-df['close']).shift(-1)
df['Close-High']=(df["close"]-df['high']).shift(-1)
df['Close-Low']=(df["close"]-df['low']).shift(-1)

df.dropna(inplace=True)
y = df[['signal']].copy()
X = df[[ 'rsi','adx', 'corr','volume','Open-Close','Close-Low']].copy()

**Comment by hsalahi23** [comment]
> The code is using future data in the model training features,".shift(-1)" which is leading to look ahead bias: df['Open-Close']=(df["open"]-df['close']).shift(-1) df['Close-High']=(df["close"]-df['high']).shift(-1) df['Close-Low']=(df["close"]-df['low']).shift(-1)
> 
> df.dropna(inplace=True) y = df[['signal']].copy() X = df[[ 'rsi','adx', 'corr','volume','Open-Close','Close-Low']].copy()

I replaced .shift(-1) with .shift(1) (as supposed to be) and the accuracy dropped to less than 60%. Such a shame.

**Comment by AzazHassankhan** [comment]
Thank you for your thorough and insightful comments regarding our paper:
 “A performance comparison of machine learning models for stock market prediction with novel
 investment strategy” [DOI: 10.1371/journal.pone.0286362].

 We understand the concern raised regarding the use of .shift(-1) in the creation of features
 such as Open-Close, Close-High, and Close-Low, which appears to introduce a form of data
 leakage by utilizing future price data. However, we would like to offer the following
 clarifications and justifications:

 **1. Nature of the Study: Simulation-Based Strategy Evaluation**
 Our intention in this work was not to create a deployable real-time predictive model for
 trading. Rather, our goal was to evaluate the relative effectiveness of machine learning
 models under a controlled, backtested environment — effectively simulating trading strategies
 under ideal conditions to compare performance metrics such as cumulative return, Sharpe
 ratio, and drawdown.

 The use of .shift(-1) aligns features with the trade execution on the following day, thereby
 simulating scenarios where trade signals are generated using the most recent full session of
 market activity. This setup is analogous to batch processing systems, where trade signals are
 executed with known delay (next-day open based on end-of-day data).

 **2. Intentional Feature Alignment, Not Oversight**
 The implementation of .shift(-1) was a conscious decision made to align the feature vector
 with the appropriate market outcome — essentially creating a mapping between the price
 action of day t+1 and the signal that would hypothetically be generated at the end of day t.
 For example, in:
 df['Open-Close'] = (df["open"] - df['close']).shift(-1)
 the shifted difference reflects the opening momentum of the following day, a valuable signal
 when simulating strategies that enter positions at day t+1 based on information available at
 the close of day t. This does not imply that the model has access to future data in real time,
 but rather that the strategy operates with delayed execution, as would occur in end-of-day
 signal-based systems.

 **3. Reframing the Model Objective**
 While the paper uses terminology like “prediction,” this is in the context of retrospective
 classification and strategy benchmarking, not forward-looking real-time prediction. We
 acknowledge that clearer language should have been used to differentiate between
 forecasting and post-hoc strategy evaluation.
 We are open to revising the paper to:
 More clearly define the simulation context and data alignment,
 Avoid language that implies real-time forecasting,
 Include an explicit disclaimer that .shift(-1) is valid only in a simulation setting and not
 applicable to live trading systems.

**4. Consistency Across Experiments**
 All models were trained and evaluated using the same methodology and feature definitions.
 Therefore, the comparative findings across ML models remain valid, even if the reported
 absolute metrics are not applicable to a real-time trading scenario.
 In conclusion, we respectfully maintain that the methodology employed is valid within the
 scope of the study, which focuses on strategy simulation and model comparison under
 backtest conditions. Nevertheless, we are willing to revise the manuscript to clearly frame the
 study as a post-event evaluation framework, clarify our assumptions, and correct any
 language that may inadvertently suggest live prediction capability.
